app/js/components/maintenance/maintenanceCategorySelect/maintenanceCategorySelect.directive.js

```diff
@@ -40,6 +45,7 @@ function maintenanceCategorySelectCtrl($rootScope,
 
     $scope.select = function (option) {
         $scope.selected = option;
+        setUrl();
```

Comment:
this can be shortened to just:

if (option === 'All') {
      delete $location.search().maintenanceCategory.id;
}